src/Picasso_ParticleInit.hpp

```diff
@@ -37,16 +37,20 @@ struct InitRandom
 template <class CreationView, class ParticleAoSoA, class ExecutionSpace>
 void filterEmpties( const ExecutionSpace& exec_space,
                     const int local_num_create,
+                    const int previous_num_particles,
                     const CreationView& particle_created,
-                    ParticleAoSoA& particles )
+                    ParticleAoSoA& particles, const bool shrink_to_fit = true )
 {
     using memory_space = typename CreationView::memory_space;
 
     // Determine the empty particle positions in the compaction zone.
     int num_particles = particles.size();
+    // int max_local_num_create = previous_num_particles + local_num_create;
     Kokkos::View<int*, memory_space> empties(
         Kokkos::ViewAllocateWithoutInitializing( "empties" ),
-        std::min( num_particles - local_num_create, local_num_create ) );
+        std::min( num_particles - previous_num_particles - local_num_create,
+                  local_num_create ) );
```

Comment:
isn't the `size of empties` always `num_particles - previous_num_particle - local_num_create` ? Maybe I need to understand better the `empty` logic below...

Reply:
This is only to reshuffle data. So if there are more created than empty (first is smaller) we only need to iterate over the number created to back fill; if there are more empty than created, we just need to move the few created to the beginning of the list. But this should only be adding in that there could be previous particles in the list, not the overall algorithm

Reply:
I think you said this backwards? But I think I understand your point.